jenkins/opensearch-ruby/Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+            name: 'RELEASE_BRANCH',
+            description: 'Branch name to release',
+            trim: true
+        )
+    }
+    stages {
+        stage('ruby-build-sign-upload') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    if (params.RELEASE_BRANCH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('RELEASE_BRANCH parameter is not set')
+                    }
+                    git url: 'https://github.com/opensearch-project/opensearch-ruby.git', branch: "$RELEASE_BRANCH"
+                    withCredentials([string(credentialsId: 'opensearchproject-rubygems-private-key', variable: 'RUBYGEMS_PRIVATE_KEY')]) {
+                        writeFile file: 'private_key.pem', text: RUBYGEMS_PRIVATE_KEY
+                    }
+                    sh'''
+                        keypath=$(realpath private_key.pem)
+                        for f in $(find . -name '*.gemspec');\\
```

Comment:
gemspecs are code and many projects inspects if the private key exists and then builds with signing, [example](https://github.com/NiketaJen/CLI-app-project-week-3/blob/54a4c477729a23a0a08cc67ce9eb215b115d088a/vendor/bundle/ruby/2.6.0/gems/wisper-2.0.1/wisper.gemspec#L20-L25).

This code should not update files in the repository, instead the gemfiles should attempt to read from an agreed location `~/.ssh/gem-private_key.pem` seems fine, then this job can put the key at that location.